gallery_ripper.py

```
@@ -3588,9 +3826,30 @@ def do_discover(self, url, quick):
                 self.yandex_results = tree_data.get("yandex_results", [])
                 self.after(0, lambda results=self.yandex_results: self.show_yandex_preview(results))
             self.after(0, lambda: self.log("Discovery complete! (cached & partial refreshed if needed)"))
+        except YandexBlockedError as e:
+            self.after(0, lambda err=e: self.handle_yandex_blocked(err))
         except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)